lib/pf/UnifiedApi/Controller/Violations.pm

```diff
@@ -21,33 +21,45 @@ has dal => 'pf::dal::violation';
 has id_key => 'id';
 has resource_id => 'id';
 
+sub create {
+    my ($self) = @_;
+    my $data = $self->parse_json;
+    if(pf::violation::violation_exist_open($data->{mac}, $data->{vid})) {
+        return $self->render(status => 409, json => { status => 409, message => $self->status_to_error_msg(409) });
+    }
+    my $response = pf::violation::violation_add($data->{mac}, $data->{vid}, (
+        status       => $data->{status},
+        start_date   => $data->{start_date},
+        release_date => $data->{release_date},
+        notes        => $data->{notes},
+        ticket_ref   => $data->{ticket_ref}
+    ));
+    $data->{id} = $response;
+    return $self->render(status => 201, json => { status => 201, data => $data, message => "violation created" }) if $data->{id};
+    return $self->render(status => 400, json => { status => 400, message => $self->status_to_error_msg(400) });
+}
 
-sub list {
+sub read_list {
     my ($self) = @_;
     my @violations = pf::violation::violation_view_open_uniq();
```

Comment:
As commented in the other branch, this subroutine shouldn't be overriden and should use the result coming from the CRUD controller since we don't only want the open violations